python/vowpalwabbit/pyvw.py

```diff
@@ -2117,28 +2215,28 @@ def from_colnames(cls, y, x, df, label_type="simple_label"):
         dict_label_type = {
             "simple_label": SimpleLabel,
             "multiclass": MulticlassLabel,
+            "multilabels": MultiLabels,
         }
 
-        if isinstance(y, list):
+        if label_type not in dict_label_type:
+            raise ValueError(
+                "'label_type' should be either of the following string: {label_types}".format(
+                    label_types=repr(list(dict_label_type.keys()))[1:-1]
+                )
+            )
+
+        if isinstance(y, list) and label_type != "multilabels":
             if len(y) == 1:
                 y = y[0]
             else:
                 raise ValueError(
                     "Argument 'y' should a list of one string (or a string)."
                 )
-        if not isinstance(y, str):
+        if not isinstance(y, str) and label_type != "multilabels":
```

Comment:
on a first glance, this `if` 'wants' to be an else if, but I noticed that it shouldn't. Maybe add a space for readability before line 2235.

Reply:
do we need to add another check to see if label_type is multilabel and the list doesn't contain strings?

Reply:
You're right this part is awkward: you could provided a list of non string for multilabels and no error would be raised. 

I think the testing for string should happen right after testing for appropriate label type. What do you think ?